airflow/providers/qdrant/hooks/qdrant.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any
+
+from grpc import RpcError
+from qdrant_client import QdrantClient
+from qdrant_client.http.exceptions import UnexpectedResponse
+
+from airflow.hooks.base import BaseHook
+
+
+class QdrantHook(BaseHook):
+    """
+    Hook for interfacing with a Qdrant instance.
+
+    :param conn_id: The connection id to use when connecting to Qdrant. Defaults to `qdrant_default`.
+    """
+
+    conn_name_attr = "conn_id"
+    conn_type = "qdrant"
+    default_conn_name = "qdrant_default"
+    hook_name = "Qdrant"
+
+    @classmethod
+    def get_connection_form_widgets(cls) -> dict[str, Any]:
+        """Returns connection widgets to add to connection form."""
+        from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import BooleanField, IntegerField, StringField
+
+        return {
+            "url": StringField(
+                lazy_gettext("URL"),
+                widget=BS3TextFieldWidget(),
+                description="Optional. Qualified URL of the Qdrant instance."
+                "Example: https://xyz-example.eu-central.aws.cloud.qdrant.io:6333",
+                default=None,
+            ),
+            "grpc_port": IntegerField(
+                lazy_gettext("GPRC Port"),
+                widget=BS3TextFieldWidget(),
+                description="Optional. Port of the gRPC interface.",
+                default=None,
+            ),
+            "prefer_gprc": BooleanField(
+                lazy_gettext("Prefer GRPC"),
+                widget=BS3TextFieldWidget(),
+                description="Optional. Whether to use gPRC interface whenever possible in custom methods.",
+                default=True,
+            ),
+            "https": BooleanField(
+                lazy_gettext("HTTPS"),
+                widget=BS3TextFieldWidget(),
+                description="Optional. Whether to use HTTPS(SSL) protocol.",
+            ),
+            "prefix": StringField(
+                lazy_gettext("Prefix"),
+                widget=BS3TextFieldWidget(),
+                description="Optional. Prefix to the REST URL path."
+                "Example: `service/v1` will result in http://localhost:6333/service/v1/{qdrant-endpoint} for REST API.",
+            ),
+            "timeout": IntegerField(
+                lazy_gettext("Timeout"),
+                widget=BS3TextFieldWidget(),
+                description="Optional. Timeout for REST and gRPC API requests.",
+                default=None,
+            ),
+        }
+
+    @classmethod
+    def get_ui_field_behaviour(cls) -> dict[str, Any]:
+        """Returns custom field behaviour."""
+        return {
+            "hidden_fields": ["schema", "login", "extra"],
+            "relabeling": {"password": "API Key"},
+        }
+
+    def __init__(self, conn_id: str = default_conn_name, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.conn_id = conn_id
+        self.get_conn()
```

Comment:
Why do you call this method here?

Reply:
Missed removing it. Great catch, thank you.
https://github.com/apache/airflow/pull/36805/commits/b97418434021ca335ae8ddf8c71f374c2a78a78a